### PR TITLE
Add join example.

### PR DESCRIPTION
**Pull request by stbof**
DOC-3191

### CODE REVIEW

**Review by yuzefovich — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14603#-:-N7Wd1TD2m9snVWGLUky:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @stbof)

File: v22.1/explain-analyze.md [hunk 1]
```diff
@@ -212,6 +212,51 @@ For example, the following `EXPLAIN ANALYZE` statement executes a simple query a
 (30 rows)
 ~~~
 
+If you perform a join, the estimated max memory allocation is also reported for the join. For example:
```
Comment (taroface) [inline v22.1/explain-analyze.md:215]: Perhaps call out the `maximum memory usage` field and value in the following example? Or maybe just the value, if we think the field names might be subject to change.

**Review by taroface — APPROVED** [review]
One suggestion for clarity; otherwise LGTM.

File: v22.1/explain-analyze.md (hunk 1 above)
Comment (stbof) [reply v22.1/explain-analyze.md:215]: The example is demonstrating that `estimated max memory allocated` was added to the `lookup join` component of the `EXPLAIN ANALYZE` output.

**Pull request merged**